osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/mapping/PushSetupAlarmConverter.java

```diff
@@ -34,23 +40,48 @@
     if (source == null) {
```

Comment:
Could source be null without having a hasSendDestinationAndMethod or hasPushObjectList?

Reply:
The null check is for safety, to avoid an NPE while converting.
In case source is not null, both SendDestinationAndMethod and PushObjectList are optional, so they might not be present.